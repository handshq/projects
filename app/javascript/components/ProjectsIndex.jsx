import React, { useEffect } from "react";
import axios from "axios";

export default function ProjectsIndex() {
  // useEffect(() => {
  //   axios.get("/projects.json").then((response) => {
  //     console.log(response.data)
  //   })
  // }, [])

  const date = new Date().toDateString();

  const projects = [
    { id: "1", name: "Dummy 0", start_date: date, end_date: date },
    { id: "2", name: "Dummy 1", start_date: date, end_date: date },
    { id: "3", name: "Dummy 2", start_date: date, end_date: date },
    { id: "4", name: "Dummy 3", start_date: date, end_date: date },
    { id: "5", name: "Dummy 4", start_date: date, end_date: date }
  ];

  return (
    <>
      <h1>Projects</h1>

      <div className="projects">
        <table className="projects-filters">
          <thead>
            <tr>
              <th>Filters</th>
            </tr>
          </thead>
          <tbody>
            <tr><td></td></tr>
            <tr><td></td></tr>
            <tr><td></td></tr>
            <tr><td></td></tr>
          </tbody>
        </table>
        <table className="projects-list">
          <thead>
            <tr>
              <th>Name</th>
              <th>Start date</th>
              <th>End date</th>
            </tr>
          </thead>
          <tbody>
            {projects.map((project) => {
              return (
                <tr key={project.id}>
                  <td>{project.name}</td>
                  <td>{project.start_date}</td>
                  <td>{project.end_date}</td>
                </tr>
              )
            })}
          </tbody>
        </table>
      </div>
    </>
  )
}
