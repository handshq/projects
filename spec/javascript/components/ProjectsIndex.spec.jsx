import React from "react";
import renderer from "react-test-renderer";
import sinon from "sinon";

import ProjectsIndex from "../../../app/javascript/components/ProjectsIndex";

let server;

const mockDate = new Date(0);

describe(ProjectsIndex, () => {
  beforeEach(() => {
    jest.spyOn(global, "Date").mockImplementation(() => mockDate)
    server = sinon.fakeServer.create();
  })

  afterEach(() => {
    jest.restoreAllMocks()
    server.restore()
  })

  it("renders correctly", () => {
    const tree = renderer.create(<ProjectsIndex />);

    expect(tree).toMatchSnapshot()
  })
})
