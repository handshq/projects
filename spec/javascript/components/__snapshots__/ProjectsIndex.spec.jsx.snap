// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsIndex renders correctly 1`] = `
[
  <h1>
    Projects
  </h1>,
  <div
    className="projects"
  >
    <table
      className="projects-filters"
    >
      <thead>
        <tr>
          <th>
            Filters
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td />
        </tr>
        <tr>
          <td />
        </tr>
        <tr>
          <td />
        </tr>
        <tr>
          <td />
        </tr>
      </tbody>
    </table>
    <table
      className="projects-list"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Start date
          </th>
          <th>
            End date
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Dummy 0
          </td>
          <td>
            Thu Jan 01 1970
          </td>
          <td>
            Thu Jan 01 1970
          </td>
        </tr>
        <tr>
          <td>
            Dummy 1
          </td>
          <td>
            Thu Jan 01 1970
          </td>
          <td>
            Thu Jan 01 1970
          </td>
        </tr>
        <tr>
          <td>
            Dummy 2
          </td>
          <td>
            Thu Jan 01 1970
          </td>
          <td>
            Thu Jan 01 1970
          </td>
        </tr>
        <tr>
          <td>
            Dummy 3
          </td>
          <td>
            Thu Jan 01 1970
          </td>
          <td>
            Thu Jan 01 1970
          </td>
        </tr>
        <tr>
          <td>
            Dummy 4
          </td>
          <td>
            Thu Jan 01 1970
          </td>
          <td>
            Thu Jan 01 1970
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
